CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -810,199 +813,249 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+        {
+            int allInBet = 0; //initiating variable that will hold all in wager
+            int numBets = playerBets.Count;
+
+            //all-in bet added to the pot, other players matched bets added to pot as well
+            allInBet = playerBets.Min();
+            AddToPot(allInBet, Enumerable.Repeat(allInBet, numBets).ToList());
+
+            //subtracting all-in bet value from all other bets, the all-in bet value has already been added to a pot
+            for (int player = 0; player < numBets; player++)
+            {
+                playerBets[player] -= allInBet; // Subtract the amount put into pot from each bet
+            }
+            currentBet -= allInBet;
+
+            //modify list of players that are eligible for the new main pot by removing player(s) that are all-in
+            List<int> allInPlayers = playerBets
+                .Select((value, index) => new { value, index })  // Keep both the value and the index
+                .Where(x => x.value == 0)  // Filter where the bet is 0 (all-in)
+                .Select(x => Pots[0].EligiblePlayers[x.index])  // Select the corresponding player from EligiblePlayers at the same index
+                .ToList();
+
+            numBets -= allInPlayers.Count;
+            playerBets.RemoveAll(value => value == 0);
+
+            //New side pot created. Pot that all-in player can win is shifted to inactive position, no more bets can be added to this pot
+            Pots.Add(new TexasHoldEmManager.TexasHoldEmPot(PotType.SIDE));
+            Pots[Pots.Count - 1].IncrementPot(Pots[0].Total); //shifting the pot that the all in player can win to the side, this pot will no longer be added to
+            Pots[0].DecrementPot(Pots[Pots.Count - 1].Total); //reseting active pot value to empty
+            Pots[Pots.Count - 1].EligiblePlayers = new List<int>(Pots[0].EligiblePlayers); //copys the list of players eligible to win the now side pot
+
+            //removing all players that are all in from eligibility from main pot and any pots created in the future
+            for (int player = 0; player < allInPlayers.Count; player++)
             {
-                return Pots.Select(pot => pot.Total).ToList();
+                Pots[0].RemoveEligiblePlayer(allInPlayers[player]);
             }
 
-            /// <summary>
-            /// Resets the pots list to an empty state.
-            /// </summary>
-            public void ResetPots()
+            //if there is a second player that has gone all in this round, recursively add pots until player that has called max bet has been reached
+            if (currentBet != playerBets.Min())
             {
-                Pots.Clear();
+                CreateSidePots(currentBet, playerBets);
+            }
+
+            //if all players but one are all-in 
+            else if (numBets == 1)
+            {
+                Pots.RemoveAt(0); //remove redundant pot creation
```

Comment:
uh why is this happening? we should remove the code that makes this redundant pot, instead of just removing it.